src/node/stage/approved.rs

```diff
@@ -902,18 +1016,95 @@ impl Approved {
             return true;
         }
 
+        if new_infos.len() > 1 {
+            debug!("splitting with new_infos {:?}", new_infos);
+        }
+
         for info in new_infos {
             let participants: BTreeSet<_> = info.elder_ids().copied().collect();
-            let _ = self.dkg_cache.insert(participants.clone(), info);
-            self.consensus_engine.vote_for(NetworkEvent {
-                payload: AccumulatingEvent::StartDkg(participants),
-                proof_share: None,
-            });
+            let parsec_version = self.consensus_engine.parsec_version();
+            let dkg_key = (participants.clone(), parsec_version);
```

Comment:
Why does the parsec version need to be a part of the dkg key?

Reply:
just to avoid the case that voting of {A,B,C} then {A,B,C,D} then {A,B,C} got messed up. 
quite rare, but put just as a safty.

Reply:
Hmm, if we can only vote a node joined or left (for simplicity) then this is OK. If D is voted joined then he is until he is voted to remove. This is the crdt part here really and we should use that. So if we had a section with ABC and were adding D then we can easily etc. 
In any case as the section chain is the agreed order of groups (each section key == a specific and unique group) then it's also cemented in place there.

Reply:
OK. Maybe better to use the current section key index instead of parsec version, to avoid making the impression that the dkg is somehow still tried to parsec. But that's a trivial thing which can be addressed anytime, doesn't have to be in this PR.

Also, this will likely get sorted when we implement the thing where we require nodes to change their name before promotion.

Reply:
Either in this PR or a FIXME in this PR? I am happy either way.

Reply:
I will put a FIXME comment in for using section_key index

> this will likely get sorted when we implement the thing where we require nodes to change their name before promotion.

will still be issue if section doesn't have enough nodes. i.e. elder_size =5, then the 5th newly added elder dropped immediately. The other four never change, hence won't have name changed, leaving `praticipants` remain the same.

Reply:
Think that's Ok @maqi  the remaining 4 will do a new dkg and therefor a new unique section key.

Reply:
That's a nice feature of a group key, It means the original A B C are not the same as A B C if D joined and left quick. So first A B C are all using different secretShares than the last A B C 
Where as if we just used ED25519 keys and counted them to get quorum then we can have confusion as both A B C seem the same.

Reply:
Yes. but this round of vote could be messed up with the previous voting round, if parsec_version/section_key not being part of the DkgKey, as the participants are the same.

Reply:
Not if the group are recognised by the section key as that's unique.  So the voting round should include the section key as the group identifier, just like the parsec version would.